docs: add list of project-ideas for AsyncAPI 2023 mentorship in project-ideas.md (#755)
asyncapi/community: mentorship/asyncapi-mentorship/2023/project-ideas.md

# Projects ideas

### Ideas list

> Number | Idea | Area | Lead mentor | Why should it be picked?

Reviewer: ```suggestion
Number | Idea | Area | Lead Mentor | Scope
```

:---:|---|:---:|:---:|---
1 | [Add help/{command} endpoint](https://github.com/asyncapi/server-api/issues/144) | [server-api](https://github.com/asyncapi/server-api) | [BOLT04](https://github.com/BOLT04) | By adding a new endpoint to the AsyncAPI server API that provides detailed information and usage instructions for the CLI tool, we can help users overcome challenges and achieve greater success in their API development efforts.
2 | [Website UI Kit design/dev project](https://github.com/asyncapi-archived-repos/design-system/issues/4) | [design-system](https://github.com/asyncapi-archived-repos/design-system) | [AceTheCreator](https://github.com/AceTheCreator) | There are a lot of places where components need to be created for even the smallest of elements that repeat on the website.

> 3 | [MVP integration of extensions catalog with AsyncAPI tools](https://github.com/asyncapi/extensions-catalog/issues/78) | [extensions-catalog](https://github.com/asyncapi/extensions-catalog) | [Lukasz Gornicki](https://github.com/derberg) | People use extensions and we should support at least the official ones in our tools.

Reviewer: ```suggestion
3 | [MVP integration of extensions catalog with AsyncAPI tools](https://github.com/asyncapi/extensions-catalog/issues/78) | [extensions-catalog](https://github.com/asyncapi/extensions-catalog) | [Lukasz Gornicki](https://github.com/derberg) | People use extensions, and we should support the official ones in our tools.
```

4 | [Documenting how different protocols work with AsyncAPI](https://github.com/orgs/asyncapi/discussions/533) | Documentation | [Alejandra Quetzalli](https://github.com/alequetzalli) |Community uses diff protocols, because everyone has diverse needs. To get more tutorials with different protocol setups. 

> 5 | [Rewrite this template and NodeJS WS template](https://github.com/asyncapi/nodejs-template/issues/133) | [nodejs-template](https://github.com/asyncapi/nodejs-template) | [Lukasz Gornicki](https://github.com/derberg) | [nodejs-template](https://github.com/asyncapi/nodejs-template) and also [nodejs-ws-template](https://github.com/asyncapi/nodejs-ws-template) are writing with old [Nunjucks templating engine](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#nunjucks). We need to rewrite them to [React templating engile](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#react-1) that should become the default and only engine in the future.

Reviewer: ```suggestion
5 | [Rewrite this template and NodeJS WS template](https://github.com/asyncapi/nodejs-template/issues/133) | [nodejs-template](https://github.com/asyncapi/nodejs-template) | [Lukasz Gornicki](https://github.com/derberg) | [nodejs-template](https://github.com/asyncapi/nodejs-template) and also [nodejs-ws-template](https://github.com/asyncapi/nodejs-ws-template) are written with old [Nunjucks templating engine](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#nunjucks). We need to rewrite them to [React templating engile](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#react-1) that should become the default and only engine in the future.
```

6 | [Simulator Desktop Application](https://github.com/asyncapi/community/issues/691) | [simulator](https://github.com/asyncapi/simulator) | [Nektarios Fifes](https://github.com/NektariosFifes) |To turn the cli application into a full desktop application that aims to visualize an asyncApi file and allow the user to generate traffic in an interactive way. 

> 7 | [Tutorial document or separate guides for glee](https://github.com/asyncapi/glee/issues/431) | [glee](https://github.com/asyncapi/glee) | [Souvik](https://github.com/Souvikns) & [Khuda Dad Nomani](https://github.com/KhudaDad414)|Currently, it is hard to get started with Glee as a new user, we need some kind of documentation that talks about:how to install and setup Glee,achieve what we are trying to do (details on what tweaks we can make)and how to deploy glee.

Reviewer: ```suggestion
7 | [Tutorial document or separate guides for Glee](https://github.com/asyncapi/glee/issues/431) | [Glee](https://github.com/asyncapi/glee) | [Souvik](https://github.com/Souvikns) & [Khuda Dad Nomani](https://github.com/KhudaDad414)|Currently, it is hard to get started with Glee as a new user, we need some kind of documentation that talks about how to install and setup Glee, achieve what we are trying to do (details on what tweaks we can make)and how to deploy Glee.
```

8 | [Add support for translations](https://github.com/asyncapi/website/issues/267) | [website](https://github.com/asyncapi/website) | [Maciej Urbańczyk](https://github.com/magicmatatjahu) |To enable adoption of the spec to a larger group. Translation of docs was an enabler for Kubernetes and Vue, so could be for AsyncAPI.

> 9 | [Prepare CLI for v1.0.0 release](https://github.com/asyncapi/cli/issues/599) | [cli](https://github.com/asyncapi/cli) | [Souvik](https://github.com/Souvikns) |We are going to release our first stable version of CLI as v1 and we need some help with it like:reach 90% test coverage,enable cross-platform installation- (only windows chocolatey),Documentation update - (update usage docs for all the commands with examples),Simple Integration tests to prove the binary works.

Reviewer: ```suggestion
9 | [Prepare CLI for v1.0.0 release](https://github.com/asyncapi/cli/issues/599) | [cli](https://github.com/asyncapi/cli) | [Souvik](https://github.com/Souvikns) |We are going to release our first stable version of CLI as v1. We need some help reaching 90% test coverage and enabling cross-platform installation- (only Windows chocolatey). Documentation update - (updating usage docs for all the commands with examples) and simple Integration tests to prove the binary works.
```

10 | [DocsUI: Messages Object output](https://github.com/asyncapi/asyncapi-react/issues/618) | [asyncapi-react](https://github.com/asyncapi/asyncapi-react) | [Missy Turco](https://github.com/mcturco) & [Fran Méndez](https://github.com/fmvilas) |For the design progress and proposal of the UX/UI updates to the Messages Object documentation output.

> 11 | [DocsUI: UX/UI updates](https://github.com/asyncapi/asyncapi-react/issues/617) | [asyncapi-react](https://github.com/asyncapi/asyncapi-react) | [Missy Turco](https://github.com/mcturco) & [Fran Méndez](https://github.com/fmvilas) |UX/UI updates to the documentation components like sidebar navigation,Servers Object output..

Reviewer: ```suggestion
11 | [DocsUI: UX/UI updates](https://github.com/asyncapi/asyncapi-react/issues/617) | [asyncapi-react](https://github.com/asyncapi/asyncapi-react) | [Missy Turco](https://github.com/mcturco) & [Fran Méndez](https://github.com/fmvilas) |UX/UI updates to the documentation components like sidebar navigation, Servers Object output.
```

12 | [Developing a Step-by-Step Contributing Guide for Newcomers](https://github.com/orgs/asyncapi/discussions/689#discussioncomment-5822686) | Documentation | |By providing clear and concise instructions, documentation, and resources, we aim to streamline the onboarding process and empower new contributors to quickly and successfully make their initial contribution.
13 | [Integration of AsyncAPI Tools Examples within AsyncAPI Tools Dashboard Tab](https://github.com/orgs/asyncapi/discussions/689#discussioncomment-5853246) |  | | To integrate the setup and examples of AsyncAPI tools into the AsyncAPI Tools Dashboard page to provide users with a comprehensive overview of each tool's functionality, setup requirements, and use cases.

> 14 | [Open Graph link preview image according to the document to open](https://github.com/asyncapi/studio/issues/224)  | [studio](https://github.com/asyncapi/studio) | |Whenever a link to Studio (with or without those query params) is pasted into social media, the preview image says nothing about the file being shared.So we should be able to dynamically generate the preview image based on the file being shared, for example, the title, description and some stats could be shown.

Reviewer: ```suggestion
14 | [Open Graph link preview image according to the document to open](https://github.com/asyncapi/studio/issues/224)  | [Studio](https://github.com/asyncapi/studio) | |Whenever a link to Studio (with or without those query params) is pasted into social media, the preview image says nothing about the file being shared. So we should be able to dynamically generate the preview image based on the file being shared; for example, the title, description, and some stats could be shown.
```

> 15 | ["Open with Studio" button for Markdown files](https://github.com/asyncapi/studio/issues/218)  | [studio](https://github.com/asyncapi/studio) | |To design a button that says something like "Open with Studio", with the AsyncAPI logo somewhere that people can use for linking to AsyncAPI files in their Markdown/HML files.

Reviewer: ```suggestion
15 | ["Open with Studio" button for Markdown files](https://github.com/asyncapi/studio/issues/218)  | [Studio](https://github.com/asyncapi/studio) | |To design a button that says something like "Open with Studio", with the AsyncAPI logo somewhere that people can use for linking to AsyncAPI files in their Markdown/HML files.
```

> 16 | [Enhanced Collaboration and Real-time Editing](https://github.com/orgs/asyncapi/discussions/689#discussioncomment-5899110)  | [studio](https://github.com/asyncapi/studio) | |To enhance collaboration and real-time editing capabilities within AsyncAPI Studio by adding features like conflict detection and resolution mechanisms,user presence indicators,version history capabilities.

Reviewer: ```suggestion
16 | [Enhanced Collaboration and Real-time Editing](https://github.com/orgs/asyncapi/discussions/689#discussioncomment-5899110)  | [Studio](https://github.com/asyncapi/studio) | |To enhance collaboration and real-time editing capabilities within AsyncAPI Studio by adding features like conflict detection and resolution mechanisms, user presence indicators, and version history capabilities.
```

> 17 | [API Development Platform Integration](https://github.com/asyncapi/studio/issues/620)  | [studio](https://github.com/asyncapi/studio) | |To integrate AsyncAPI Studio with API development platforms to enhance the overall development experience.Features can be developed:API Platform Integration,Security and Authentication,API Documentation Generation,Deployment and Testing.

Reviewer: ```suggestion
17 | [API Development Platform Integration](https://github.com/asyncapi/studio/issues/620)  | [Studio](https://github.com/asyncapi/studio) | |To integrate AsyncAPI Studio with API development platforms to enhance the overall development experience. Features can be developed: API Platform Integration, Security and Authentication, API Documentation Generation, Deployment, and Testing.
```

18 | [Library for easier integration testing of code generators](https://github.com/asyncapi/generator/issues/752)  | [generator](https://github.com/asyncapi/generator) | |To have feature in Generator or maybe a separate library that we can easily enable in generator repository and enable integration tests for template.
19 | [Enable Metric Monitoring and Observability Modules for Code Generators](https://github.com/orgs/asyncapi/discussions/689#discussioncomment-5960921)  | [generator](https://github.com/asyncapi/generator) | |Metric monitoring helps collect and analyze the data generated from an application and can be vital when it comes to maintaining the progress of a system. While AsyncAPI’s current code generator produces a well-formed structure to be built on, this project proposes to develop a feature that can allow users to easily accommodate a metric monitoring module within their application.

> 20 | [Drag & drop AsyncAPI block builder](https://github.com/asyncapi/studio/issues/265)  | [studio](https://github.com/asyncapi/studio) | |Currently we have in the studio Visualiser which renders the flow of operations in an application described using AsyncAPI.But it only works in readonly mode.So good solution would be to add the ability to create AsyncAPI specifications from scratch (or update existing one) using drag and drop blocks, which could be connected, similar to the current blocks in visualiser.

Reviewer: ```suggestion
20 | [Drag & drop AsyncAPI block builder](https://github.com/asyncapi/studio/issues/265)  | [Studio](https://github.com/asyncapi/studio) | |Currently, we have in the Studio Visualiser which renders the flow of operations in an application described using AsyncAPI. But it only works in read-only mode. So a good solution would be to add the ability to create AsyncAPI specifications from scratch (or update existing ones) using drag and drop blocks, which could be connected, similar to the current blocks in the visualizer.
```
